### DISCUSSION TITLE
authOptions are being imported client side.

### DISCUSSION THREAD

**Discussion by GravyPouch** [nextauthjs/next-auth#5303] [Help]
### Question 💬

I am trying to use "unstable_getServerSession". The [guide ](https://next-auth.js.org/configuration/nextjs )online shows to import authOptions. Every time I do I receive the following error: `Module not found: Can't resolve 'fs'`. I assume this is because Next.js is trying to use this on the client-side. Which obviously won't work. I can't figure out what I am doing wrong, I am sure it is something silly. thank you.

### How to reproduce ☕️

This is the code I am trying to call authOptions from:
```

import { Dropdown, Badge } from "@nextui-org/react";
import { FaBell } from "react-icons/fa";
import { authOptions } from "../pages/api/auth/[...nextauth]";
import { unstable_getServerSession } from "next-auth/next";

export default function Notify({ session }) {
  return (
    <Dropdown placement="bottom-left">
      <Dropdown.Trigger>
        <div>
          <Badge color="primary" content={badgeCount}>
            <FaBell size={"1.5em"} />
          </Badge>
        </div>
      </Dropdown.Trigger>
      <Dropdown.Menu color="secondary" aria-label="Notifications">
        <Dropdown.Item key="Notifications">Notifications</Dropdown.Item>
      </Dropdown.Menu>
    </Dropdown>
  );
}

export async function getServerSideProps(context) {
  const session = await unstable_getServerSession(
    context.req,
    context.res,
    authOptions
  );

  if (!session) {
    return {
      redirect: {
        destination: "/",
        permanent: false,
      },
    };
  }

  return {
    props: {
      session,
    },
  };
}

```

This is what my [...nextauth] file looks like:

```
import NextAuth from "next-auth";
import { PrismaAdapter } from "@next-auth/prisma-adapter";
import DiscordProvider from "next-auth/providers/discord";
import EmailProvider from "next-auth/providers/email";
import prisma from "../../../lib/prisma";

export const authOptions = {
  adapter: PrismaAdapter(prisma),
  session: { strategy: "jwt" },

  providers: [
    EmailProvider({
      server: {
        host: process.env.EMAIL_SERVER_HOST,
        port: process.env.EMAIL_SERVER_PORT,
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD,
        },
      },
      from: process.env.EMAIL_FROM,
    }),
    DiscordProvider({
      clientId: process.env.DISCORD_CLIENT_ID,
      clientSecret: process.env.DISCORD_CLIENT_SECRET,
    }),
  ],
};

export default NextAuth(authOptions);
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
This is likely a different issue. Next.js eliminates code from the client bundle that is used in data fetching methods like `getServerSideProps`.

You can check what is being bundled for the client here: https://next-code-elimination.vercel.app/s/OFaFof__B

In your code, I don't see where `badeCount` comes from, seems to be not declared, maybe something there?

Not related, but your import

`import { authOptions } from "../pages/api/auth/[...nextauth]";`

can just be `import { authOptions } from "./api/auth/[...nextauth]";`

**Comment by smithmilner** [comment]
I'm having the same problem using `_app.tsx` `getInitialProps` https://next-code-elimination.vercel.app/s/tTYxoUqSo

It shows as:

```
error - ./node_modules/openid-client/lib/helpers/deep_clone.js:1:33
Module not found: Can't resolve 'v8'

Import trace for requested module:
./node_modules/openid-client/lib/issuer.js
./node_modules/openid-client/lib/index.js
./node_modules/next-auth/core/lib/oauth/callback.js
./node_modules/next-auth/core/routes/callback.js
./node_modules/next-auth/core/routes/index.js
./node_modules/next-auth/core/index.js
./node_modules/next-auth/next/index.js
./node_modules/next-auth/index.js
./src/pages/_app.tsx

https://nextjs.org/docs/messages/module-not-found
```

The docs mention I should be able to use unstable_getServerSession inside getInitialProps here https://next-auth.js.org/getting-started/client#sessionprovider

Any guidance would be appreciated

**Reply by GravyPouch** [reply]
I get the same "v8" not found error if I remove the {authOptions} import but keep the import for getServerSideAuth

**Comment by 4lirexa** [comment]
same problem
